### ISSUE TITLE
How to use post_process function in a custom plugin to catch events?

### ISSUE THREAD

**Issue by anjoshua** [getsentry/self-hosted#2264]
### Self-Hosted Version

23.6.2 (and 21.11.0)

### CPU Architecture

x86_64

### Docker Version

20.10.22

### Docker Compose Version

2.15.1

### Steps to Reproduce

1. Make a new plugin (based on Redmine plugin). Add the following for testing if post_process is getting called:
    ```
    def post_process(self, group, event, is_new, **kwargs):
      self.logger.warning("TESTING: Received a new event in plugin")
     ```
    I am trying to override the [post_process](https://github.com/getsentry/sentry/blob/master/src/sentry/plugins/base/v1.py#L454) function in base Plugin class.

2. Install plugin, add to project, configure plugin

3. Send new events (confirm `sentry.tasks.post_process.post_process_group` is being triggered in Sentry admin panel queue page)

### Expected Result

I am expecting 
`docker compose logs | grep TESTING`
or
`docker compose logs | grep <plugin __name__>`
to show the test warning that I added in the plugin, so that I can confirm the above `post_process` function is getting called

### Actual Result

There is no log :(

I can log in the `__init__` function and see that the plugin is being initialised, but can't see the log I added in `post_process`.

I originally was using 21.11.0, but also tested in 23.6.2 in case it was a version issue.
In version 21.11.0, In the queue page of the Admin panel, I can see `sentry.tasks.post_process.post_process_group` is being triggered when I send new events, but `sentry.tasks.post_process.plugin_post_process_group` is not. In 23.6.2 there is no `sentry.tasks.post_process.plugin_post_process_group` option on that queue page.

I am wondering, what are the requirements for `post_process` to be called in a plugin? From looking a the code (mostly for version 21.11.0), I thought that when a new event was processed by `sentry.tasks.post_process.post_process_group`, it would check all active plugins on the project and call `post_process` for each plugin. Are there some other requirements that need to be met?

**Comment by hubertdeng123** [comment]
I'm unfamiliar with how this works generally but is this a custom integration that you have built? How are you installing the plugin, is it something like the instructions here?
https://github.com/getsentry/integration-platform-example/#using-your-integration

**Comment by anjoshua** [comment]
> I'm unfamiliar with how this works generally but is this a custom integration that you have built? How are you installing the plugin, is it something like the instructions here? https://github.com/getsentry/integration-platform-example/#using-your-integration

Hmm it doesn't look like that...

It is a modified version of the built-in redmine plugin https://github.com/getsentry/sentry/tree/master/src/sentry_plugins/redmine

It is a plugin written by a former employee in my company (and they were using Sentry 21.11.0)

But I have figured it out! The issue was how I was installing the plugin (thank you for asking me how I was installing the plugin, as that is how I realised). And now it works in both 21.11.0 and 23.6.2.

I was trying to install it using `requirements.txt` in sentry folder, but it was not working. So instead I was installing it manually using `docker compose exec web /bin/bash -c "pip install -U /etc/sentry/<plugin>.tar.gz" && docker compose restart web relay nginx` (which was written in some documentation by the former employee, I am not sure where they got it from). But this just installs the plugin in the `web` container, and the `post_process` task is in the `worker` container, so I guess that is why it didn't work. 

In 21.11.0, I am not sure why using` requirements.txt` wasn't working, but I updated `endpoint.sh` with the above `pip install` line and that works now. For 23.6.2, I can use `requirements.txt`, but instead of `/etc/sentry/<plugin>.tar.gz` it seems I have to use `/usr/src/sentry/<plugin>.tar.gz `. After installing like this, I can now see the plugin is catching events and logging in both versions. 

It seems like it was my fault for installing it in a non-standard way.. Sorry about that 😅
